src/mixins/filing-template-mixin.ts

```diff
@@ -141,46 +142,30 @@ export default class FilingTemplateMixin extends Mixins(DateMixin, EnumMixin) {
       // make a copy so we don't change original array
       let parties = cloneDeep(this.getOrgPeople)
 
-      // add completing party (client error correction only)
+      // add completing party (client error corrections only)
       if (this.isClientErrorCorrection) {
         parties = this.addCompletingParty(parties)
       }
 
+      // prepare parties
+      parties = isDraft ? parties : this.prepareParties(parties)
```

Comment:
Bug fix: this extra function wasn't previously called for firm corrections. Now it is.

(Prepare Parties removes any parties deleted in this filing, which actually isn't supported in firm corrections, but this is common code here and it's safe.)